app/core/db.py

```
@@ -65,9 +64,27 @@ def getconn():
-    engine = sqlalchemy.create_engine(db_url, pool_pre_ping=True)
+    # 개별 값 기반 DSN
+    if not (settings.DB_USER and settings.DB_PASSWORD and settings.DB_NAME):
+        raise DatabaseConfigError()
+
+    host = getattr(settings, "DB_HOST", None) or "localhost"
+    port = int(getattr(settings, "DB_PORT", None) or 5432)
+
+    SUPPORTED_DRIVERS = {"psycopg", "pg8000", "psycopg2"}
+    driver = getattr(settings, "DB_DRIVER", "psycopg")  # psycopg 또는 pg8000
+    if driver not in SUPPORTED_DRIVERS:
+        raise DatabaseConfigError(f"지원하지 않는 DB_DRIVER: {driver}. 지원 드라이버: {SUPPORTED_DRIVERS}")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)